book/src/leader-leader-handoff.md

```diff
@@ -0,0 +1,101 @@
+# Leader to Leader Handoff
+
+This design describes how leader's handoff production of the PoH ledger between
+each other as each leader generates its own slot.
+
+## Challenges
+
+Current leader and the next leader are both racing to generate the final tick
+for the current slot.  The next leader may arrive at that slot while still
+processing the current leader's entries.
+
+The ideal scenario would be that the next leader generated its own slot right
+after it was able to vote for the current leader.  It is very likely that the
+next leader will arrive at their PoH slot height before the current leader
+finishes broadcasting the entire block.
+
+The next leader has to make the decision of attaching its own block to the last
+completed block, or wait to finalize the pending block.  It is possible that the
+next leader will produce a block that proposes that the current leader failed,
+even though the rest of the network observes that block succeeding.
+
+The current leader has incentives to start its slot as early as possible to
+capture economic rewards.  Those incentives need to be balanced by the leader's
+need to attach its block to a block that has the most commitment from the rest
+of the network.
+
+## Leader timeout
+
+While a leader is actively receiving entries for the previous slot, the leader
+can delay broadcasting the start of its block in real time.  The delay is
+locally configurable by each leader, and can be dynamically based on the
+previous leader's behavior.  If the previous leader's block is confirmed by the
+leader's TVU before the timeout, the PoH is reset to the start of the slot and
+this leader produces its block immediately.
+
+The downsides:
+
+* Leader delays its own slot, potentially allowing the next leader more time to
+catch up.
+
+The upsides compared to guards:
+
+* All the space in a block is used for entries.
+
+* The timeout is not fixed.
+
+* The timeout is local to the leader, and therefore can be clever.  The leader's
+heuristic can take into account avalanche performance.
+
+* This design doesn't require a ledger hard fork to update.
+
+* The previous leader can redundantly transmit the last entry in the block to
```

Comment:
what if we changed Broadcast/Avalanche to send every Blob to the next leader (as well as spreading it around the network)?

Reply:
@rob-solana,  double the bandwidth. 

The next leader should only pick the previous block if the rest of the network will pick it as well.  Actually seeing the retransmission of the blobs is a pretty good signal of how far the packets are getting.